Assets/Scripts/UI/InGame/HUD/Weapon/Blades.cs

```diff
@@ -1,17 +1,66 @@
 ﻿using UnityEngine;
+using UnityEngine.UI;
 
 namespace Assets.Scripts.UI.InGame.Weapon
 {
     public class Blades : MonoBehaviour
     {
+        [HideInInspector]public float bladeSta = 100f;
+        [HideInInspector]public float curGas = 100f;
+
         public GameObject BladeLeft;
         public GameObject BladeRight;
+        public GameObject currentLeftBlade;
+        public GameObject currentRightBlade;
 
         public GameObject BladeLeftSprite;
         public GameObject BladeRightSprite;
-        private int distance = 18;
+        public GameObject currentLeftBladeSprite;
+        public GameObject currentRightBladeSprite;
+
+        public GameObject bloodPrefab;
+
+        public UnityEngine.Sprite[] BladesSprite;
+
+        public Image leftImage;
+        public Image rightImage;
+        
+        private GameObject curLeft;
+        private GameObject curRight;
+
         private int previousBlades;
 
+        private void Update()
+        {   
+            if(leftImage != null && rightImage != null)
```

Comment:
Invert if statement to make it easier to understand. Might want to use a ScriptableObject array as well here, similar to the aforementioned DamageColors